### PR TITLE
(Feature) Known abis

### LINKED ISSUE #48: Known ABIs
It would be nice if eth-cli would come with certain well-known ABIs (ERC20, ERC721, etc.). I see two use cases for this:

- Use them in any command that expects a path to an ABI file (`contract:load`, etc.)
- Add a command that prints the ABI. For example, if you want to add an ERC20 token to metamask, doing `eth abi:show ERC20` would be slightly more convenient than googling it.

### PR DESCRIPTION
**Pull request by Agupane**
Closes #48 

## Description
Adds support for listing well-known abis like `ERC20 & ERC721`

### The new commands are:
- `eth abi ABI_NAME`: shows the abi of the `ABI_NAME`
- `eth abi list`: displays the list of available abis

### Supported abis:
- `ERC20`
- `ERC721`

Also if you want to load the contract on different methods like `eth contract:load`, you can use the `ABI_NAME` instead of an abi object.

### Example:
`eth contract:load ERC20 0x121345`

ERC20.json is taken from [here](https://github.com/buhrmi/non-fungible-token-abi/blob/master/mintable-non-fungible-token.abi.json)
ERC721.json is taken from [here](https://github.com/ethereum/wiki/wiki/Contract-ERC20-ABI)

### CODE REVIEW

File: src/commands/abi.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { Command, flags } from '@oclif/command'
+
+import { getAbiByName } from '../helpers/abi/knowAbis'
+
+export default class AbiCommand extends Command {
+  static description = 'Displays the abi of a given well-know abi (ERC20, ERC721)'
```
Comment (fvictorio) [inline src/commands/abi.ts]:
```suggestion
  static description = 'Displays a known ABI (ERC20, ERC721)'
```

File: src/commands/abi.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+  }
+
+  static args = [
+    {
+      name: 'abi',
+      required: true,
+      description: 'The contract abi name.',
```
Comment (fvictorio) [inline src/commands/abi.ts]:
```suggestion
      description: 'The contract name.',
```

File: src/commands/abi.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  static examples = ['eth abi ERC20', 'eth abi ERC721']
+
+  async run() {
+    const { args } = this.parse(AbiCommand)
+    const { abi } = args
+    let abiStr: string | null = getAbiByName(abi)
+    if (abiStr) {
+      this.log(abiStr)
+    } else {
+      this.error(`Abi for ${abi} not found!`)
```
Comment (fvictorio) [inline src/commands/abi.ts]:
```suggestion
      this.error(`ABI for ${abi} not found!`)
```

**Comment by fvictorio** [comment]
Maybe we could have a `eth abi:show ERC20` command and then a `eth abi:list` command that shows the list of known ABIs.

**Comment by Agupane** [comment]
It's done now @fvictorio

**Comment by fvictorio** [comment]
Two more things:

- When showing an ABI, use an styled JSON, like it's done in the `networks` command.
- When listing the known ABIs, just output one per line. This makes it easier to process (in case it's needed, no matter how unlikely it seems) and if someone wants to know what exactly that list is, they can use `eth abi:list --help`.

**Comment by fvictorio** [comment]
Also, please add a couple of e2e tests.

**Comment by Agupane** [comment]
It's done @fvictorio

**Pull request merged**